api/circulation.py

```diff
@@ -717,3 +717,64 @@ def default_notification_email_address(self, patron, pin):
         of changes?
         """
         return Configuration.default_notification_email_address()
+
+
+    def checkin(self, patron, pin, licensepool):
+        """  Return a book early.  
+
+        :param patron: a Patron object for the patron who wants
+        to check out the book.
+        :param pin: The patron's alleged password.
+        :param licensepool: Identifier of the book to be checked out is 
+        attached to this licensepool.
+        """
+        raise NotImplementedException
+
+
+    def checkout(self, patron, pin, licensepool, internal_format):
+        """Check out a book on behalf of a patron.
+
+        :param patron: a Patron object for the patron who wants
+        to check out the book.
+        :param pin: The patron's alleged password.
+        :param licensepool: The Identifier of the book to be checked out is 
+        attached to this licensepool.
+        :param internal_format: Represents the patron's desired book format.
+
+        :return: a LoanInfo object.
+        """
+        raise NotImplementedException
+
+
+    def fulfill(self, patron, pin, licensepool, internal_format):
+        """ Get the actual resource file to the patron.
+        TODO:  to the patron?  or to some intermediary stage?
```

Comment:
This method's job is to create a FulfillmentInfo object; it will be delievered immediately to the patron, but it's not the repsonsibility of this method to care about that.